### PR TITLE
Server: Extended OpenAPI support

### LINKED ISSUE #12: Support additional OpenAPI features
The zswag-client should support additional OpenApi features to enable more REST-like interfaces:
- [x] #16 Support PUT/DELETE/PATCH __[Released with 0.6.0]__
- [x] #18 Support x-zserio-request-part=* __[Released with 0.6.0]__
- [x] #20 Support parameter schema formats (string/hex/byte/binary)
- [x] #17 Support value array passing schemes
- [x] #19 Support x-zserio-request-part with request field name
- [x] #15 Support path-based argument passing (to support GET)

Comprehensively formulated, this would enable method specs like this:
```yaml
paths:
  <Path>
    [get|put|post|patch|delete]:
       operationId: <Service-Function>
       parameters:
         - in: [query|path]
           name: <Name>
           x-zserio-request-part: <[*|field-name]>
           style: [matrix|form|simple]
           explode: [true|false]
           schema:
             format: [string|byte|hex|binary]
       requestBody:
         content:
           application/x-zserio-object
```

### LINKED ISSUE #15: Support path-based argument passing
* [x] Client Side
* [x] Server Side

### LINKED ISSUE #17: Support value array passing schemes
* [x] Client Side
* [x] Server Side

### LINKED ISSUE #19: Support x-zserio-request-part with request field name
* [x] Client Side
* [x] Server Side

### LINKED ISSUE #20: Support parameter schema formats (string/hex/byte/binary)
The following formats should be supported 
* `string`: Default string representation of the x-zserio-request-part field (only works for native types).
* `hex`: Hex representation of the x-zserio-request-part field.
* `byte` | `base64`: Base64 representation of the x-zserio-request-part field.
* `base64url`: Base64url representation of the x-zserio-request-part field.
* `binary`: Binary representation of the x-zserio-request-part field.

Tasks:
* [x] Client Side
* [x] Server Side

### PR DESCRIPTION
**Pull request by josephbirkner**
### Changes

Closes #12 
Closes #15 
Closes #17 
Closes #19
Closes #20

### Review Checklist

- [x] The changes are understood.
- [x] The changes are well documented.
- [x] The changes have been tested.

### CODE REVIEW

**Review by KRE02 — CHANGES_REQUESTED** [review]

File: libs/zswagcl/include/zswagcl/http-service.hpp [hunk 1]
```diff
@@ -149,7 +149,7 @@ struct HTTPService::Config
          *
          * Ignored if HTTP-Method is GET.
          */
-        bool bodyRequestObject = true; // TODO: Not used
+        bool bodyRequestObject = false; // TODO: Not used
```
[the remark this answers is not in the document]
Comment (josephbirkner) [reply libs/zswagcl/include/zswagcl/http-service.hpp:152]: Just for testing while waiting for #29

File: libs/zswag/test/test_integration.bash [hunk 2]
```diff
@@ -34,6 +34,7 @@ while [[ $# -gt 0 ]]; do
     -b|--background)
       echo "→ Launching $2 ..."
       $2 &
+      sleep 10
```
Comment (KRE02) [inline libs/zswag/test/test_integration.bash:37]: Why is this needed?

Comment (josephbirkner) [reply libs/zswag/test/test_integration.bash:37]: The server startup can take between 1-10 seconds. Normally it's <5, but in one case on macOS it took longer than 5.

File: libs/zswag/test/client.cpp [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    auto testCounter = 0;
+    auto failureCounter = 0;
+
+    std::cout << "Starting integration tests with " << specUrl << "\n";
+
+    auto runTest = [&] (
+        std::function<zsr::Introspectable(ZsrClient&)> const& fn,
+        double expect,
+        std::string const& aspect
+    ) mutable -> void
```
Comment (KRE02) [inline libs/zswag/test/client.cpp:31]: Nitpick: remove `mutable`

**Review by KRE02 — APPROVED** [review]

**Pull request merged**